src/main/java/slimeknights/tconstruct/library/recipe/casting/ContainerFillingRecipe.java

```diff
@@ -0,0 +1,99 @@
+package slimeknights.tconstruct.library.recipe.casting;
+
+import lombok.Getter;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.Items;
+import net.minecraft.item.crafting.IRecipeSerializer;
+import net.minecraft.item.crafting.IRecipeType;
+import net.minecraft.util.IItemProvider;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.world.World;
+import net.minecraftforge.fluids.FluidStack;
+import net.minecraftforge.fluids.FluidUtil;
+import net.minecraftforge.fluids.capability.IFluidHandler;
+import net.minecraftforge.fluids.capability.IFluidHandlerItem;
+import slimeknights.tconstruct.library.recipe.RecipeTypes;
+import slimeknights.tconstruct.smeltery.TinkerSmeltery;
+import slimeknights.tconstruct.smeltery.recipe.ICastingInventory;
+
+public abstract class ContainerFillingRecipe implements ICastingRecipe {
+  @Getter
+  protected final IRecipeType<?> type;
+  @Getter
+  protected final ResourceLocation id;
+  @Getter
+  protected final String group;
+  @Getter
+  protected final int fluidAmount;
+  @Getter
+  protected final IItemProvider result;
```

Comment:
Rename this to container as well, don't like comparing item to to a field named `result` when its the input.

Could also use `Item` here instead of `IItemProvider`, fie, so there is no need to call `asItem()` each time.